pkg/controller/sparkapplication/controller.go

```diff
@@ -617,6 +617,17 @@ func (c *Controller) submitSparkApplication(app *v1beta2.SparkApplication) *v1be
 		}
 	}
 
+	// Use batch scheduler to perform scheduling task before submitting (before build command arguments).
```

Comment:
So this is moved up so the annotations can be passed to the driver/executor pods, right?

Reply:
yes, you are right.